### ISSUE TITLE
Warning "file_exists(): open_basedir restriction in effect" installing in a shared server

### ISSUE THREAD

**Issue by GustavoMarcante** [contentacms/contenta_jsonapi#88]
Hi. Congratulations for the awesome project of Contenta!
I installed Contenta in my desktop (Ubuntu 16.04) without problems, but installing in a Linux shared server I got this warning after "Configure site" and in all the pages (where USER is my account):

```
Warning: file_exists(): open_basedir restriction in effect. File(/) is not within the allowed path(s): (/home/USER/:/tmp:/usr/local/php/7.0/lib/php:./) in League\OAuth2\Server\CryptKey->__construct() (line 43 of /home/USER/app/contenta/vendor/league/oauth2-server/src/CryptKey.php).

League\OAuth2\Server\CryptKey->__construct('file://') (Line: 50)
League\OAuth2\Server\ResourceServer->__construct(Object, NULL) (Line: 42)
Drupal\simple_oauth\Server\ResourceServer->__construct(Object, Object, Object, Object) (Line: 280)
Drupal\Component\DependencyInjection\Container->createService(Array, 'simple_oauth.server.resource_server') (Line: 177)
Drupal\Component\DependencyInjection\Container->get('simple_oauth.server.resource_server', 1) (Line: 494)
Drupal\Component\DependencyInjection\Container->resolveServicesAndParameters(Array) (Line: 236)
Drupal\Component\DependencyInjection\Container->createService(Array, 'simple_oauth.authentication.simple_oauth') (Line: 177)
Drupal\Component\DependencyInjection\Container->get('simple_oauth.authentication.simple_oauth', 1) (Line: 494)
Drupal\Component\DependencyInjection\Container->resolveServicesAndParameters(Array) (Line: 329)
Drupal\Component\DependencyInjection\Container->createService(Array, 'authentication_collector') (Line: 177)
Drupal\Component\DependencyInjection\Container->get('authentication_collector') (Line: 72)
Drupal\jsonapi\Routing\Routes::create(Object) (Line: 28)
Drupal\Core\DependencyInjection\ClassResolver->getInstanceFromDefinition('\Drupal\jsonapi\Routing\Routes') (Line: 123)
Drupal\Core\Controller\ControllerResolver->createController('\Drupal\jsonapi\Routing\Routes::entryPoint') (Line: 74)
Drupal\Core\Controller\ControllerResolver->getControllerFromDefinition('\Drupal\jsonapi\Routing\Routes::entryPoint') (Line: 145)
Drupal\Core\Routing\RouteBuilder->rebuild() (Line: 83)
Drupal\Core\ProxyClass\Routing\RouteBuilder->rebuild() (Line: 1141)
drupal_flush_all_caches() (Line: 2360)
install_config_import_batch_finish(1, Array, Array, '2 min 21 sec') (Line: 414)
_batch_finished() (Line: 81)
_batch_page(Object) (Line: 642)
install_run_task(Array, Array) (Line: 550)
install_run_tasks(Array) (Line: 121)
install_drupal(Object) (Line: 44)
```

I don't have access to my php.ini and I cannot change "open_basedir restriction" through the server PHP options.
Any idea about how to disable this? Or is it possible to fix Contenta (as it do not happens in another Drupal distributions)?
Thanks.

**Comment by dawehner** [comment]
Thank you for reporting your problem!

Do you mind trying out https://www.drupal.org/node/2890894#comment-12149528 ? I think this should fix the particular problem you have.

**Comment by e0ipso** [comment]
Is it possible that after deploying this the path to your encryption files is not accessible? Did you configure Simple OAuth in your local and then moved the configuration upward?

**Comment by GustavoMarcante** [comment]
Thanks for the replies.

@e0ipso I don't know if this is your question, but (remembering that is a shared server), I
installed Contenta in ~/app/contenta and
created a link from ~/www/contenta to ~/app/contenta/web
The owner or all files is my user (not www-data as in my Ubuntu desktop).
I did not move files or directories.

@dawehner Sorry for my question, but I'm new in Drupal and I never applied a patch.
I downloaded https://www.drupal.org/files/issues/2890894-2.patch to contenta directory.
I tried
`patch < 2890894-2.patch`
and I got
`can't find file to patch at input line 5`
Looking for the patch file, I supposed the file to change is web/modules/contrib/simple_oauth/src/Server/ResourceServer.php, I informed this to patch, but I got
```
Hunk #1 FAILED at 37.
1 out of 1 hunk FAILED -- saving rejects to file web/modules/contrib/simple_oauth/src/Server/ResourceServer.php.rej
```
The same message I had trying with:
```
cd web/modules/contrib/simple_oauth/src/Server
patch < ../../../../../../2890894-2.patch
```
I tried to change this this file **manually** (deleting the four lines and including the six lines), and:
**it worked!!!**

Thanks.
